src/Widgets/MultitaskingView.vala

```diff
@@ -46,6 +46,24 @@ namespace Gala
 		Actor workspaces;
 		Actor dock_clones;
 
+		static int animation_freeze_count = 0;
+		public static void freeze_animations ()
+		{
+			animation_freeze_count++;
+		}
+
+		public static void thaw_animations ()
+		{
+			animation_freeze_count--;
+
+			assert (animation_freeze_count >= 0);
+		}
```

Comment:
Why is this counting system necessary? Shouldn't there only be one drag end handler active at the same time?

Reply:
This is just some future-proofing so that if more objects use this, it won't break. This kind of thing is also used in other classes such as `WorkspaceManager`.